Clarify degree formatting (#3048)
modelica/ModelicaSpecification: chapters/unitexpressions.tex

unit-symbol :
   unit-char { unit-char }

unit-char :
   NON-DIGIT
\end{lstlisting}

> A \lstinline[language=grammar]!unit-symbol! is a non-empty string of \lstinline[language=grammar]!unit-char! characters.

Reviewer: Sorry, my suggestion was to also remove this sentence.  It is redundant now.

Author: Ok!

The set of allowed characters is currently not defined, but it is recommended to only use \lstinline[language=grammar]!NON-DIGIT!.
It is required that basic and derived units of the SI system are recognized, but tools are allowed to additionally support user-defined unit symbols.
The required unit symbols do not make use of Greek letters, but a unit such as $\Omega$ is spelled out as \lstinline!"Ohm"!.
Similarly degree is spelled out as \lstinline!"deg"!, both on its own (for angles) and as part of \lstinline!"degC"!, \lstinline!"degF"! and \lstinline!"degRk"! for temperatures (Celsius, Fahrenheit and Rankine).

A \lstinline[language=grammar]!unit-operand! should first be interpreted as a \lstinline[language=grammar]!unit-symbol! and only if not successful the second alternative assuming a prefixed operand should be exploited.
There must be no spacing between the \lstinline[language=grammar]!unit-symbol! and a possible \lstinline[language=grammar]!unit-prefix!.
The values of the prefixes are according to the ISO standard.
The letter \lstinline!u! is used as a symbol for the prefix \emph{micro}.

\begin{nonnormative}

> A tool may present \lstinline!"Ohm"! as $\Omega$ and the prefix \lstinline!"u"! as $\mu$, and similarly \lstinline!"m2"! as m\textsuperscript{2}.
> In addition a tool may present \lstinline!"deg"! as $^{\circ}$, both on its own (for angles) and for temperatures - e.g., \lstinline!"degC"! can be written as $^{\circ}$C.

Reviewer: En-dash, say _presented_ for consistency, and match style of the added content with style of pre-existing rules:
```suggestion
A tool may present \lstinline!"Ohm"! as $\Omega$ and the prefix \lstinline!"u"! as $\mu$.
Exponents such as \lstinline!"m2"! may be presented as m\textsuperscript{2}.
Degrees may be presented as $^{\circ}$, both for \lstinline!"deg"! on its own (for angles) and for temperatures -- e.g., \lstinline!"degC"! can be presented as $^{\circ}$C.
```

\end{lstlisting}

A \lstinline[language=grammar]!unit-symbol! is a non-empty string of \lstinline[language=grammar]!unit-char! characters.

> These are restricted to \lstinline[language=grammar]!NON-DIGIT! characters.

Reviewer: Wouldn't it be better to say this directly in the unit string grammar, thereby also incorporating the sentence above?

Author: Good idea, done.

> The set of allowed characters is currently not defined, but it is recommended to only use \lstinline[language=grammar]!NON-DIGIT!.

Author: ```suggestion
These are restricted to \lstinline[language=grammar]!NON-DIGIT! characters.
```

\end{lstlisting}

A \lstinline[language=grammar]!unit-symbol! is a non-empty string of \lstinline[language=grammar]!unit-char! characters.

> The set of allowed characters is currently not defined, but it is recommended to only use \lstinline[language=grammar]!NON-DIGIT!.

Reviewer: I find this too weak.  If we should ever be able to build a working solution for unit checking, there cannot be uncertainty about counts as a valid unit symbol.

Reviewer: I think this would be a good topic the upcoming phone meeting.  It would also be interesting to get input from others interested in unit checking, such as @casella, @perost, @jespermattsson.

Author: Poll (unit should be SI-unit, this is primarily displayUnit):
1. No change: Hans, Martin
2. Recommended NON-DIGIT: Gerd, Hans
3. Strictly NON-DIGIT: Henrik, Christoff, Markus, Martin, Elena.
=> Strict

It is required that basic and derived units of the SI system are recognized, but tools are allowed to additionally support user-defined unit symbols.
The required unit symbols do not make use of Greek letters, but a unit such as $\Omega$ is spelled out as \lstinline!"Ohm"!.

> Similarly degree is spelled out as \lstinline!deg! for angles, and \lstinline!degC!, \lstinline!degF! and \lstinline!degRk! for temperatures (Celsius, Fahrenheit and Rankine).

Reviewer: It seems like overspecification to restrict this to angles and temperature.  Shouldn't it be a statement just about symbols, similarly to `"Ohm"`?

Reviewer: There are also missing string quotes:
```suggestion
Similarly degree is spelled out as \lstinline!"deg"! for angles, and \lstinline!"degC"!, \lstinline!"degF"! and \lstinline!"degRk"! for temperatures (Celsius, Fahrenheit and Rankine).
```

Author: I tried to explain that it is more generic for degree. Good?

\begin{nonnormative}
A tool may present \lstinline!Ohm! as $\Omega$ and the prefix \lstinline!u! as $\mu$, and similarly \lstinline!m2! as m\textsuperscript{2}.

> In addition a tool may present \lstinline!deg! as $^{\circ}$ (for angle) and \lstinline!degC! as $^{\circ}$C (for temperature).

Reviewer: This isn't generic enough to cover `"degF"`.

Author: I don't see that it is needed as it is non-normative, and I find the next line more important and that one is only relevant for the SI-standard, and thus not for Fahrenheit. Explaining that would be too much text.

Author: But I tried to add a bit more.

Reviewer: Looks good now.  Resolving.
